pgjdbc/src/main/java/org/postgresql/util/PSQLState.java

```diff
@@ -118,12 +122,29 @@ public String getState() {
     return this.state;
   }
 
-  public static boolean isConnectionError(@Nullable String psqlState) {
-    return PSQLState.CONNECTION_UNABLE_TO_CONNECT.getState().equals(psqlState)
-        || PSQLState.CONNECTION_DOES_NOT_EXIST.getState().equals(psqlState)
-        || PSQLState.CONNECTION_REJECTED.getState().equals(psqlState)
-        || PSQLState.CONNECTION_FAILURE.getState().equals(psqlState)
-        || PSQLState.CONNECTION_FAILURE_DURING_TRANSACTION.getState().equals(psqlState);
+  private static final Collection<PSQLState> CONNECTION_ERRORS = Arrays.asList(
+    CONNECTION_UNABLE_TO_CONNECT,
+    CONNECTION_DOES_NOT_EXIST,
+    CONNECTION_REJECTED,
+    CONNECTION_FAILURE,
+    CONNECTION_FAILURE_DURING_TRANSACTION);
+  private static final Map<String, PSQLState> CODE_TO_ENUM;
+
+  static {
+    final PSQLState[] values = PSQLState.values();
```

Comment:
In practice Java has limit of 2700 or so values in a single enum: https://stackoverflow.com/questions/4468388/maximum-number-of-enum-elements-in-java

PostgreSQL defines ~270 error codes, so it looks we are safe, and all of them would fit into the enum: https://github.com/postgres/postgres/blob/ff50baec65bba1a839322ba8bcb3efcd08f40621/src/backend/utils/errcodes.txt

On the other hand, we might want to extract interface for the SqlState.